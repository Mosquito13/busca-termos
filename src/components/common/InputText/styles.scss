@import '../../../sass/styles.scss';

.input-text {
    flex: 1;
    display: flex;
    height: 4rem;

    &__input {
        flex: 1;
        height: 4rem;
        min-width: 4rem;
        border: 1px solid currentColor;
        outline: none;
        padding: 0 1rem;
        background-color: transparent;
        color: currentColor;
        font-family: inherit;
        font-size: inherit;
        transition: all .2s;
        border-radius: $radius;
        cursor: text;

        &::placeholder {
            @include themify() {
                color: themed('color-text-disabled');
            }
        }

        &:hover,
        &:focus,
        &:active {
            box-shadow: 0px 0px 5px currentColor;
        }

        &--disabled {
            box-shadow: none;
            @include themify() {
                border-color: themed('color-text-disabled');
            }

            &:hover,
            &:focus,
            &:active {
                box-shadow: none;
            }
        }

        &--borderless {
            border: none;
            border-radius: 0;
        }
    }
}
