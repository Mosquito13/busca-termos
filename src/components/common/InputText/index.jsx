import { useCallback, forwardRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './styles.scss';

const InputText = forwardRef(
  ({ placeholder, value, onChange, disabled, borderless }, ref) => {
    const classes = classNames(
      'input-text__input',
      disabled && 'input-text__input--disabled',
      borderless && 'input-text__input--borderless'
    );

    const handleChange = useCallback((event) => onChange(event.target.value), [onChange]);

    return (
      <div className="input-text">
        <input
          ref={ref}
          className={classes}
          type="text"
          value={value}
          placeholder={placeholder}
          onChange={handleChange}
          disabled={disabled}
          spellCheck={false}
        />
      </div>
    );
  }
);

InputText.propTypes = {
  /**
   * Placeholder do campo
   */
  placeholder: PropTypes.string,
  /**
   * Valor do campo
   */
  value: PropTypes.string,
  /**
   * Funcao para alterar o valor do campo
   */
  onChange: PropTypes.func,
  /**
   * Indica se esta desabilitado
   */
  disabled: PropTypes.bool,
  /**
   * Indica se deve renderizar sem bordas
   */
  borderless: PropTypes.bool
};

InputText.defaultProps = {
  placeholder: '',
  value: ''
};

InputText.name = 'InputText';

export default InputText;
