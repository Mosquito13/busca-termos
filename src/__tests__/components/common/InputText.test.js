import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import InputText from '../../../components/common/InputText';

describe(`${InputText.name}`, () => {
  describe('tests', () => {
    let wrapper, onChangeSpy;

    beforeAll(() => {
      onChangeSpy = jest.fn();

      wrapper = shallow(
        <InputText
          value=""
          onChange={onChangeSpy}
          placeholder="A placeholder"
        />
      );
    });

    it('should render', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.input-text')).toHaveLength(1);
      expect(wrapper.find('.input-text__input')).toHaveLength(1);
    });

    it('should call onChange', () => {
      wrapper
        .find('.input-text__input')
        .simulate('change', { target: { value: 'newValue' } });
      expect(onChangeSpy).toBeCalledTimes(1);
      expect(onChangeSpy).toBeCalledWith('newValue');
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render without modifier', () => {
      const tree = renderer.create(<InputText />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render borderless', () => {
      const tree = renderer.create(<InputText borderless />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render disabled', () => {
      const tree = renderer.create(<InputText disabled />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
