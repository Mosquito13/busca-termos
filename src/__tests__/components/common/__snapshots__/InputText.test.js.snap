// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText snapshots should render borderless 1`] = `
<div
  className="input-text"
>
  <input
    className="input-text__input input-text__input--borderless"
    onChange={[Function]}
    placeholder=""
    spellCheck={false}
    type="text"
    value=""
  />
</div>
`;

exports[`InputText snapshots should render disabled 1`] = `
<div
  className="input-text"
>
  <input
    className="input-text__input input-text__input--disabled"
    disabled={true}
    onChange={[Function]}
    placeholder=""
    spellCheck={false}
    type="text"
    value=""
  />
</div>
`;

exports[`InputText snapshots should render without modifier 1`] = `
<div
  className="input-text"
>
  <input
    className="input-text__input"
    onChange={[Function]}
    placeholder=""
    spellCheck={false}
    type="text"
    value=""
  />
</div>
`;
